adapters/rtbhouse/rtbhouse.go

```diff
@@ -156,13 +161,63 @@ func (adapter *RTBHouseAdapter) MakeBids(
 	for _, seatBid := range openRTBBidderResponse.SeatBid {
 		for _, bid := range seatBid.Bid {
 			bid := bid // pin! -> https://github.com/kyoh86/scopelint#whats-this
-			typedBid = &adapters.TypedBid{Bid: &bid, BidType: "banner"}
-			bidderResponse.Bids = append(bidderResponse.Bids, typedBid)
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			} else {
+				typedBid = &adapters.TypedBid{
+					Bid:     &bid,
+					BidType: bidType,
+				}
+
+				// for native bid responses fix Adm field
+				if typedBid.BidType == openrtb_ext.BidTypeNative {
+					bid.AdM, err = getNativeAdm(bid.AdM)
+					if err != nil {
+						errs = append(errs, err)
+						continue
+					}
+				}
+
+				bidderResponse.Bids = append(bidderResponse.Bids, typedBid)
+			}
 		}
 	}
 
 	bidderResponse.Currency = BidderCurrency
 
-	return bidderResponse, nil
+	return bidderResponse, errs
+
+}
+
+func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
+	switch bid.MType {
+	case openrtb2.MarkupBanner:
+		return openrtb_ext.BidTypeBanner, nil
+	case openrtb2.MarkupNative:
+		return openrtb_ext.BidTypeNative, nil
+	default:
+		return "", fmt.Errorf("unrecognized bid type in response from rtbhouse for bid %s", bid.ImpID)
+	}
+}
+
+func getNativeAdm(adm string) (string, error) {
+	nativeAdm := make(map[string]interface{})
+	err := json.Unmarshal([]byte(adm), &nativeAdm)
+	if err != nil {
+		return adm, errors.New("unable to unmarshal native adm")
+	}
+
+	// move bid.adm.native to bid.adm
+	if _, ok := nativeAdm["native"]; ok {
+		//using jsonparser to avoid marshaling, encode escape, etc.
+		value, dataType, _, err := jsonparser.Get([]byte(adm), string(openrtb_ext.BidTypeNative))
+		if err != nil || dataType != jsonparser.Object {
+			return adm, errors.New("unable to get native adm")
+		}
+		adm = string(value)
```

Comment:
In this block of code, `[]byte(adm)` is unMarshaled and then jsonparser.Get is again used to check if native key exist in adm object.

we could use `jsonparser.Get` here.   The `jsonparser.Get` will error if native  is not in adm

```
func getNativeAdm(adm string) (string, error) {
  value, dataType, _, err := jsonparser.Get([]byte(adm), string(openrtb_ext.BidTypeNative))
  if err != nil || dataType != jsonparser.Object {
	return adm, errors.New("unable to get native adm")
  }
  return string(value), nil
}
```

Reply:
@onkarvhanumante See pubmatic's function https://github.com/prebid/prebid-server/blob/a6dea2f26d0dcf9e20c30de8297dad8931a319ee/adapters/pubmatic/pubmatic.go#L497 
The purpose of the function is to replace `bid.adm` with what optionally `bid.adm.native` would be. This is due to the fact that older Native specification placed the native response in `bid.adm.native` but the current spec has it in `bid.adm` directly. We need to abide by the latest specs. Parsing the external response is risky, so `jsonparser.Get()` may result with error.
If `bid.adm` after being parsed doesn't contain `native` key then this function will return `bid.adm` without any modifications.
I would not touch this part. Your modification does not take into account that `bid.adm` may as well not contain `native` which is also good. 
The first unmarshalling has the purpose of checking if the string in `bid.adm` is a well defined JSON string.
The second one reads what `bid.adm.native` would contain, if present (if not that's fine, we do not change `bid.adm` then). It will only error if the dataType of the parsed string is not an object (it must be as per specs).